lib/cronitor/monitor.rb

```diff
@@ -0,0 +1,190 @@
+# frozen_string_literal: true
+
+def symbolize_keys(obj)
+  obj.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}
+end
+
+module Cronitor
+  class Monitor
+    attr_reader :key, :data, :api_key, :api_version, :env
+
+    @@HEADERS = {
+      'Content-Type': 'application/json',
+      'User-Agent': 'cronitor-ruby',
+      'Cronitor-Version': Cronitor.api_version
+    }
+
+    PING_RETRY_THRESHOLD = 5
+
+    def self.put(opts={})
+      rollback = opts[:rollback] || false
+      opts.delete(:rollback)
+
+      monitors = opts[:monitors] || [opts]
+
+      resp = HTTParty.put(
+        Cronitor.monitor_api_url,
+        basic_auth: {
+          username: Cronitor.api_key,
+          password: ''
+        },
+        body: {
+          monitors: monitors,
+          rollback: rollback
+        }.to_json,
+        headers: @@HEADERS,
+        timeout: 10
+      )
+
+      if resp.code == 200
+        _monitors = []
+        data = JSON.parse(resp.body)
+
+        (data['monitors']||[]).each do |md|
+          m = Monitor.new(md['key'])
+          m.data = symbolize_keys(md)
+          _monitors << m
+        end
+        return _monitors.length == 1 ? _monitors[0] : _monitors
+      elsif resp.code == 400
+        raise ValidationError.new(resp.body)
+      else
+        raise Error.new("Error connecting to Cronitor: #{resp.code}")
+      end
+    end
+
+    def self.delete(key)
+      resp = HTTParty.delete(
+        "#{Cronitor.monitor_api_url}/#{key}",
+        timeout: 10,
+        basic_auth: {
+          username: Cronitor.api_key,
+          password: ''
+        },
+        headers: @@HEADERS
+      )
+      if resp.code != 204
+        Cronitor.logger.error("Error deleting monitor: #{key}")
+        return false
+      end
+      true
+    end
+
+    def initialize(key, api_key: nil, api_version: nil, env: nil)
+      @key = key
+      @api_key = api_key || Cronitor.api_key
+      @env = env || Cronitor.environment
+      @@HEADERS['Cronitor-Version'] = api_version if api_version
```

Comment:
Here is a case where class vars might get tricky: if you've got multiple monitors, this value will bleed from one to another, b/c they’ll all be sharing `@@HEADERS`.